streams/upgrade-system-tests-21/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java

```diff
@@ -59,40 +59,33 @@ public static void main(final String[] args) throws Exception {
         final KafkaStreams streams = new KafkaStreams(builder.build(), config);
         streams.start();
 
-        Runtime.getRuntime().addShutdownHook(new Thread() {
```

Comment:
Java8 cleanup only

Reply:
Similar cleanup https://github.com/apache/kafka/pull/6598

Reply:
Similar cleanup in https://github.com/apache/kafka/pull/6599

Reply:
Different PR for different modules to allow corresponding cherry-picking to older branches.